extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -2304,4 +2272,72 @@ public static function onGetTalkPage( Title $title, Title &$talkPageTitle ) {
 		return true;
 	}
 
+	/**
+	 * SUS-845: Lock down Thread namespace - only allow edits to users with proper permission or to people editing their own posts
+	 * Everything else is disallowed.
+	 * @param Title $title
+	 * @param User $user
+	 * @param string $action
+	 * @param string|string[] $result string or array of strings - MediaWiki error message key(s) to be shown if error
+	 * @return bool Whether the user is allowed to perform this action or true if this is not a Wall/Forum title
+	 */
+	public static function onGetUserPermissionsErrors( Title $title, User $user, string $action, &$result ): bool {
+		global $wgWallNS;
+
+		// Wall or Forum threads
+		if ( in_array( MWNamespace::getSubject( $title->getNamespace() ), $wgWallNS ) ) {
+			$wallMessage = new WallMessage( $title );
+			$isAuthor = $wallMessage instanceof WallMessage && $wallMessage->isAuthor( $user );
+			$isActionAllowed = true;
+			switch ( $action ) {
+				case 'edit':
+					$isActionAllowed = $isAuthor || $user->isAllowed( 'walledit' );
+					$result = $isActionAllowed ?? [ 'badaccess-group0' ];
+					break;
+				case 'create':
+					// hack - we can only allow creating a page in Thread namespace
+					// if it's coming from Wall Nirvana API, otherwise we end up with a broken page
+					$isNirvanaRequest = Transaction::getAttribute( Transaction::PARAM_ENTRY_POINT ) === Transaction::ENTRY_POINT_NIRVANA;
```

Comment:
I'd rather use a less hacky solution - instead of relying on `Transaction` API (that can change and is not owned by SUS) let's set a global in a "correct" Wall creation flow and check this flag here.